synapse/storage/databases/main/end_to_end_keys.py

```diff
@@ -32,6 +34,51 @@
 
 
 class EndToEndKeyWorkerStore(SQLBaseStore):
+    def get_e2e_device_keys_for_federation_query(self, user_id: str):
+        """Get all devices (with any device keys) for a user
+
+        Returns:
+            Deferred which resolves to (stream_id, devices)
+        """
+        return self.db_pool.runInteraction(
+            "get_e2e_device_keys_for_federation_query",
+            self._get_e2e_device_keys_for_federation_query_txn,
+            user_id,
+        )
+
+    def _get_e2e_device_keys_for_federation_query(
```

Comment:
I think this was typoed?

```suggestion
    def _get_e2e_device_keys_for_federation_query_txn(
```

Reply:
whoops. yes it was. good spot.